### PR TITLE
added DMCA notice to shot page

### PR DESCRIPTION
**Pull request by dannycoates**
This adds a dmca field to the `data` table and shows a notice on the shot page when that field is not null. It could also be used for a link to specific info but is currently only used as a flag for a general message.

### CODE REVIEW

**Comment by ianb** [comment]
Something that we're planning on for new updates (that we haven't done before) is to make the code tolerant of different database versions.  So for our purposes, 6.3.0 is the current production and runs with database version 15.  We should ensure that 6.3.0 can run with database version 16, and that master can run with database version 15.  *Or*, if we can't do that, then we are supposed to release (to production) another version which can do that.  We haven't done this yet, so I don't actually know quite what the migrations should look like!

**Comment by ianb** [comment]
After discussion, we're not going to worry about the migration/database compatibility stuff for now.

**Comment by dannycoates** [comment]
> I'm pretty sure this is leaky, it'll send all the data but just hide it on render

It never actually fetches the image. The prerendered html has the block inlined. The json contains all the data, including the image url, but never fetches it. I updated the image url to 404 on blocked images (technically not "not found" but whatevs). So whatever is "leaked" isn't useful. It might be nice to refactor things in the future, but this keeps the changes pretty minimal for now. 

I went with a single `block_type` column and an enum to represent unblocked (none) and DMCA. We can add more with `ALTER TYPE shot_block_type ADD VALUE 'abuse'` for example. I didn't add a `blocked_reason` column yet because I'm not clear on its use. Is there already a place where dmca notices are hosted? If not I'd rather wait to add that column.

**Comment by dannycoates** [comment]
@ckprice screenshots

Owner sees:

screen shot 2017-04-24 at 11 05 46 am

Others see:

screen shot 2017-04-24 at 11 04 53 am

**Comment by ckprice** [comment]
@dannycoates thanks! Just passed this around to legal.

**Comment by dannycoates** [comment]
@ckprice updated. I'm not exactly sure what "generic not available page" means so I used the 404 page as is shown for any unknown shot url. cool?

creator:

screen shot 2017-04-25 at 8 38 46 am

non-creator:

screen shot 2017-04-25 at 8 42 36 am

**Comment by clouserw** [comment]
@dannycoates that's fine, thanks (cory is pto today)

File: server/db-patches/patch-15-16.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+CREATE TYPE shot_block_type AS ENUM ('none', 'dmca');
+ALTER TABLE data ADD COLUMN block_type shot_block_type DEFAULT 'none' NOT NULL;
```
Comment (ianb) [inline server/db-patches/patch-15-16.sql]: I guess I would have used NULL as not-blocked, but I suppose this is fine too.

**Pull request merged**